packages/react/src/components/Popup/Popup.tsx

```diff
@@ -124,30 +124,32 @@ function useHideSiblingNodes(props: IPopupProps, root: React.RefObject<HTMLDivEl
   const isModalOrPanel = props['aria-modal'];
 
   React.useEffect(() => {
-    const targetDocument = getDocument();
-    if (isModalOrPanel && targetDocument && root && root.current) {
-      const popupPortalNode = root.current.parentElement?.parentElement;
-      let nodesToHide: HTMLElement[] = findSiblingNodes(popupPortalNode, popupPortalNode?.parentElement);
-
-      //if popupPortalNode is not a direct child of body, its ancestor's siblings need to be hidden as well.
-      if (popupPortalNode?.parentElement !== targetDocument.body) {
-        const popupAncestorNode = findAncestorNode(root.current, targetDocument);
-        nodesToHide.concat(findSiblingNodes(popupAncestorNode, targetDocument.body));
-      }
+    if (props.enableAriaHiddenSiblings) {
```

Comment:
If we go ahead with this approach, `enableAriaHiddenSiblings` should be saved in a local const to avoid capturing a reference to the entire `props` object in the effect.